catroid/src/androidTest/java/org/catrobat/catroid/test/content/bricks/RecentBricksTest.java

```diff
@@ -0,0 +1,135 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.test.content.bricks;
+
+import android.content.Intent;
+import android.util.Log;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.io.StorageOperations;
+import org.catrobat.catroid.ui.ProjectActivity;
+import org.catrobat.catroid.uiespresso.util.rules.FragmentActivityTestRule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+
+import androidx.test.core.app.ApplicationProvider;
+
+import static org.catrobat.catroid.common.FlavoredConstants.DEFAULT_ROOT_DIRECTORY;
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.Espresso.pressBack;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.scrollTo;
+import static androidx.test.espresso.assertion.ViewAssertions.doesNotExist;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+
+public class RecentBricksTest {
+
+	private final String projectName = "recentBrickTest";
+	private final String spriteName = "testSprite";
+
+	@Rule
+	public FragmentActivityTestRule<ProjectActivity> baseActivityTestRule = new
+			FragmentActivityTestRule<>(ProjectActivity.class, ProjectActivity.EXTRA_FRAGMENT_POSITION, ProjectActivity.FRAGMENT_SPRITES);
+
+	@Before
+	public void setUp() {
+		createProject();
+		baseActivityTestRule.launchActivity(new Intent());
+	}
+
+	@Test
+	public void testCheckRecentBrickVisible() {
+		onView(withText(spriteName)).perform(click());
+		onView(withId(R.id.button_add)).perform(click());
+		onView(withText(R.string.category_control)).perform(scrollTo(), click());
+		onView(withText(R.string.brick_forever)).perform(scrollTo(), click());
+		onView(withId(android.R.id.list)).perform(click());
+		onView(withId(R.id.button_add)).perform(click());
+		onView(withText(R.string.category_recently_used)).perform(click());
+		onBrickAtPosition(1).checkShowsText(R.string.brick_forever);
+	}
+
+	@Test
+	public void testCheckNonBackgroundBricksAreHiddenForBackgroundSprites() {
+		onView(withText(spriteName)).perform(click());
+
+		onView(withId(R.id.button_add)).perform(click());
+		onView(withText(R.string.category_pen)).perform(scrollTo(), click());
```

Comment:
Unfortunately, I have seen that this test does not pass on Jenkins. The reason for this is that the emulators are quit slow. You could try to invoke `onView(withText(R.string.category_pen)).perform(waitFor(isDisplayed(), waitThreshold));` with a certain threshold, e.g. 5000ms to be sure that the view is actually displayed before clicking it.

Reply:
Thank you so much! was gawking over this error since morning 😂 will try this and see if it works on jenkins!

Reply:
Done, seems like the problem has been solved, the error is no longer visible:)